src/main/java/at/hannibal2/skyhanni/features/inventory/SackDisplay.kt

```diff
@@ -192,50 +191,51 @@ object SackDisplay {
     }
 
     private fun MutableList<Renderable>.drawOptions(totalPrice: Long) {
-        val name = SortType.entries[config.sortingType.ordinal].longName // todo avoid ordinal
-        addString("§7Sorted By: §c$name")
-
-        addSelector<SortType>(
-            " ",
-            getName = { type -> type.shortName },
-            isCurrent = { it.ordinal == config.sortingType.ordinal }, // todo avoid ordinal
+        addRenderableButton<SortingTypeEntry>(
+            label = "Sorted By",
+            current = config.sortingType,
+            getName = { it.displayName },
             onChange = {
-                config.sortingType = SortingTypeEntry.entries[it.ordinal] // todo avoid ordinals
+                config.sortingType = it
                 update(false)
             },
+            universe = SortingTypeEntry.entries,
         )
 
-        addButton(
-            prefix = "§7Number format: ",
-            getName = NumberFormat.entries[config.numberFormat.ordinal].displayName, // todo avoid ordinal
+        addRenderableButton<NumberFormatEntry>(
+            label = "Number Format",
+            current = config.numberFormat,
+            getName = { it.displayName },
             onChange = {
-                // todo avoid ordinal
-                config.numberFormat =
-                    NumberFormatEntry.entries[(config.numberFormat.ordinal + 1) % 3]
+                config.numberFormat = it
                 update(false)
             },
+            universe = NumberFormatEntry.entries,
```

Comment:
Apparently a few of my comments from earlier didn't post. `universe` is in all cases where we're using it here, and for all intents and purposes, the `*Button<...>` funs should be changeable to:

```kotlin
inline fun <reified T: Enum<T>> MutableList<Renderable>...
...
        universe: List<T> = enumValues<T>().toList(),
```